### DISCUSSION TITLE
Java 11 / 17 compile support

### DISCUSSION THREAD

**Discussion by Arashvscode** [Sketchware-Pro/Sketchware-Pro#468] [Ideas]
App old jdk plz add jdk 18~19

@JavkhlanK 
@hasrat-ali

**Comment by j-px** [comment]
Sketchware Pro doesn't ship an entire JDK. The APK only contains android.jar, as classes not available there are not available on Android devices as well.

But I get what you mean. Being able to use e.g. the `var` keyword from Java 11 would be great. Would need upgrading of both the Java compiler (the Eclipse JDT Core Batch Compiler) and d8 to support e.g. Java 11 source code though. Still same as before, no JDK should be shipped with the app though, as core library desugaring is still out of reach (in my opinion)

**Comment by Arashvscode** [comment]
> Sketchware Pro doesn't ship an entire JDK. The APK only contains android.jar, as classes not available there are not available on Android devices as well.
> 
> But I get what you mean. Being able to use e.g. the `var` keyword from Java 11 would be great. Would need upgrading of both the Java compiler (the Eclipse JDT Core Batch Compiler) and d8 to support e.g. Java 11 source code though. Still same as before, no JDK should be shipped with the app though, as core library desugaring is still out of reach (in my opinion)

Could this feature be added in the new update?  If you add we do not need miscellaneous programs

**Comment by j-px** [comment]
I just said what'd be needed for that.
